packages/tree-select/.eslintrc.js

```diff
@@ -3,5 +3,6 @@
 module.exports = {
 	rules: {
 		'import/no-extraneous-dependencies': [ 'error', { packageDir: __dirname } ],
+		'eqeqeq': [ 'error', 'always', { null: 'ignore' } ],
```

Comment:
Is this really better than just one `eslint-disable-next-line` comment at the relevant place?

Reply:
Happy to make the change; `== null` is a robust, syntactic, and safe mechanism to determine "nullish" so it seemed fine to just always allow it.

Reply:
> `== null` is a robust, syntactic, and safe mechanism to determine "nullish"

What do you think about other usages of `==`, when comparing values different than `null`? I only remember the "don't use it" commandment dating back to the Douglas Crockford era, but I never learned more deeply about the exact rationale.

If the `eqeqq` rule still makes sense most of the time, let's just use `eslint-disable` to disable specific instances.

Reply:
Oh I'm highly opposed to all usages of `==` except `== null`/`!= null` :-) basically `==` is "implicitly coerce the types, and then do `===`", and implicit coercion harms maintainability.

However, `typeof x === 'undefined' || x === null` is not more clear, or more robust, than `x == null`, which is why the `eqeqeq` rule has always had an option to ignore comparisons with null.

Reply:
> the `eqeqeq` rule has always had an option to ignore comparisons with null.

Ha! I didn't know about that option and also didn't know it's already used in our [ESLint config package](https://github.com/Automattic/wp-calypso/blob/master/packages/eslint-config-wpcalypso/index.js#L30). And also didn't check that ESLint actually reports a warning on your `!= null` statement (it doesn't)!

That means that adding an `eqeqeq` exception to `tree-select`'s ESLint config is not necessary after all. I apologize for bothering you with a problem that doesn't exist 🤦‍♂

Reply:
no problem :-) i'll remove the override and update the PR.